sdk/src/main/java/io/radar/sdk/RadarActivityManager.kt

```diff
@@ -0,0 +1,215 @@
+            return@runCatching
+        }
+       if (!isActivityUpdatesStarted) {
+            Radar.logger.d("trying to start activity updates")
+
+            val task = activityClient.requestActivityTransitionUpdates(
+                request, RadarLocationReceiver.getActivityPendingIntent(context)
+            )
+            task.addOnSuccessListener {
+                isActivityUpdatesStarted = true
+                Radar.logger.d("Activity updates started")
+            }
+
+            task.addOnFailureListener { e: Exception ->
+                Radar.logger.e("Activity updates failed to start")
+            }
+       } else {
+            Radar.logger.d("Activity updates already started")
+       }
+    }
+
+    internal fun stopActivityUpdates() {
+        if (!isActivityUpdatesStarted) {
+            return
+        }
+        activityClient.removeActivityUpdates(RadarLocationReceiver.getActivityPendingIntent(context))
+        isActivityUpdatesStarted = false
+    }
+
+    internal fun requestMotionUpdates() {
+        sensorSnapshotManager.getSensorSnapshot()
+    }
+}
+
+class SensorSnapshotManager(private val context: Context) {
+
+    private val sensorManager: SensorManager = context.getSystemService(Context.SENSOR_SERVICE) as SensorManager
+    private var accelerometerSensor: Sensor? = null
+    private var gyroscopeSensor: Sensor? = null
+    private var magnetometerSensor: Sensor? = null
+    private var accelerometerSensorEventListener: SensorEventListener? = null
+    private var gyroscopeSensorEventListener: SensorEventListener? = null
+    private var magnetometerSensorEventListener: SensorEventListener? = null
+
+    init {
+        accelerometerSensor = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER)
+        gyroscopeSensor = sensorManager.getDefaultSensor(Sensor.TYPE_GYROSCOPE)
+        magnetometerSensor = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD)
+        
+    }
+
+    fun getSensorSnapshot() {
+        if (accelerometerSensor == null) {
+            Radar.logger.i("accelerometer_sensor not available")
+        } else {
+            accelerometerSensorEventListener = object : SensorEventListener {
+                override fun onSensorChanged(event: SensorEvent) {
+                    val sensorData = event.values
+                    Radar.logger.d("accelerometerSensor data: ${sensorData.joinToString()}")
+                    val jsonObject = JSONObject().apply {
+                        put("x", sensorData[0])
+                        put("y", sensorData[1])
+                        put("z", sensorData[2])
+                    }
+                    RadarState.setLastAccelerometer(context,jsonObject)
+
+                    sensorManager.unregisterListener(this)
+                }
+
+                override fun onAccuracyChanged(sensor: Sensor, accuracy: Int) {
+                    // We don't need to do anything here
+                }
```

Comment:
Because we're not doing anything in these overrides, should we just omit them? (this and the next two)

Reply:
The stub is required to implement the interface.